controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -824,6 +825,24 @@ private Throwable getRealException(Throwable ex) {
                 timer.getElapsed()));
     }
 
+    /**
+     * Controller Service API to delete scope recursively.
+     *
+     * @param scope Name of scope to be deleted.
+     * @param requestId request id
+     * @return Status of delete scope.
+     */
+    public CompletableFuture<DeleteScopeRecursiveStatus> deleteScopeRecursive(final String scope, final long requestId) {
+        Exceptions.checkNotNullOrEmpty(scope, "scope");
+        Timer timer = new Timer();
+
+        return streamMetadataTasks.deleteScopeRecursive(scope, requestId)
+                .thenApplyAsync(status -> {
```

Comment:
Is there a possibility where `streamMetadataTasks.deleteScopeRecursive(..)` completes exceptionally? If yes we would need to handle it .

Reply:
This is handled now, thank you

Reply:
Thanks, could you please elaborate it for completeness?

Reply:
`streamMetadataTasks.deleteScopeRecursive(..)` is now handling the exceptions by calling a new method `handleDeleteScopeError`. This method checks for the type of exception and updates the status accordingly. The general status update is `FAILURE`, so now with this the `streamMetadataTasks.deleteScopeRecursive(..)` will not complete exceptionally.